### PR TITLE
Xrt 1853 branch2

### PR DESCRIPTION
**Pull request by SteveOss**

### CODE REVIEW

**Comment by SteveOss** [comment]
@BinduRao2018 Needed to do some more stuff. So closing initial branch.

File: include/iot/component.h [hunk 1]
```diff
@@ -97,6 +102,16 @@ struct iot_component_t
  */
 extern void iot_component_init (iot_component_t * component, const iot_component_factory_t * factory, iot_component_start_fn_t start, iot_component_stop_fn_t stop);
 
+/**
+ * @brief Set component callback function, called when all container components are up (started)
+ *
+ * The function to initialise the component
```
Comment (BinduRao2018) [inline include/iot/component.h:108]: Incorrect documentation. Should this be deleted?

File: src/c/component.c [hunk 2]
```diff
@@ -25,6 +25,12 @@ void iot_component_init (iot_component_t * component, const iot_component_factor
   atomic_store (&component->refs, 1u);
 }
 
+void iot_component_set_running_callback (iot_component_t * component, iot_component_running_fn_t fn)
+{
+  assert (component && fn);
+  component->running_fn = fn;
```
Comment (BinduRao2018) [inline src/c/component.c:31]: Isn't a state check necessary as this callback can be set only before the Component start?

**Review by BinduRao2018 — COMMENTED** [review]
comments in-line

File: include/iot/component.h (hunk 1 above)
Comment (SteveOss) [reply include/iot/component.h]: Have updated

File: src/c/component.c (hunk 2 above)
Comment (SteveOss) [reply src/c/component.c:31]: Really done when component is initialised, so not necessary. If you register it too late, won't be called when everything running, although might be called subsequently as can stop then re-start.

**Review by BinduRao2018 — APPROVED** [review]
LGTM

**Comment by BinduRao2018** [comment]
@SteveOss, Could you merge as I don't have permission on this branch?

**Pull request merged**